corehq/warehouse/transforms/sql/group_dim.sql

```diff
@@ -0,0 +1,26 @@
+INSERT INTO {{ group_dim }} (
+    group_id,
+    name,
+    domain,
+    case_sharing,
+    reporting,
+    group_last_modified,
+    deleted,
+    dim_last_modified,
+    dim_created_on
+)
+SELECT
+    group_id,
+    name,
+    domain,
+    case_sharing,
+    reporting,
+    group_last_modified,
+    CASE doc_type
+        WHEN 'Group' THEN false
+        WHEN 'Group-Deleted' THEN true
+    END,
```

Comment:
i've sometimes seen weird source data for some models where the -Deleted suffix gets applied twice in some docs. i think we may want to use an else clause for all of our deleted flags